File: Python3/프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    for i in range(len(sizes)):
        w, h = sizes[i]
        if w < h:
            sizes[i] = [h, w]
    
    mx = max(s[0] for s in sizes)
    my = max(s[1] for s in sizes)
    
    answer = mx * my
    return answer

File: Java/프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        for (int i = 0; i < sizes.length; i++) {
            if (sizes[i][0] < sizes[i][1]) {
                int x = sizes[i][0];
                sizes[i][0] = sizes[i][1];
                sizes[i][1] = x;
            }
        }
        
        int maxX = 0;
        int maxY = 0;
        for (int i = 0; i < sizes.length; i++) {
            maxX = Math.max(maxX, sizes[i][0]);
            maxY = Math.max(maxY, sizes[i][1]);
        }
        
        return maxX * maxY;
    }
}